### ISSUE TITLE
Broken responsive sidebar design in categories

### ISSUE THREAD

**Issue by NinoKrb** [KrbDevelopment/Articly-Frontend#8]
**Describe the bug**
The Responsive design of the categories page does not work properly when the device is too small. The sidebar content is simply displayed under the categories without following a specific layout.

**To Reproduce**
1. Go on any category page
2. Scale your browser window down
3. Scroll down to the end of the Page
4. See the broken layout parts

**Expected behavior**
A descriptive footer which displays the content in an orderly and decent way.

**Screenshots**
![Example Screenshot](https://user-images.githubusercontent.com/66381853/131220237-db6dc8b8-f47f-4c0e-a072-c031b7579af4.png)

**Desktop (please complete the following information):**
 - OS: Ubuntu 21.04
 - Browser: Google Chrome
 - Version: 92.0.4515.107

**Comment by yamanidev** [comment]
I'd like to contribute. Please assign the issue to me.

**Comment by JanKrb** [comment]
Contribution guidelines are not ready yet, please fork the repository and create a pull request as soon as you're done. 😄

**Comment by yamanidev** [comment]
Sure thing, thanks.

**Comment by yamanidev** [comment]
Can you please guide me through the installation process? 
I read `cd Frontend` but I cannot find that folder, same for `cd Backend`.

**Comment by JanKrb** [comment]
For sure, the `cd Frontend` and `cd Backend` is still from the old repositiory KrbDevelopment/Wiki-Software. You will need to do:

Backend:
```bash
git clone https://github.com/KrbDevelopment/Articly-Backend
cd Articly-Backend
```

Frontend:
```bash
git clone https://github.com/KrbDevelopment/Articly-Frontend
cd Articly-Frontend
```

The following commands like `npm i` or `npm run build` will be the same.

Please let me know, if you need further help. 😄

**Comment by yamanidev** [comment]
I have built the project, and it is running on my local machine. I am at the login page, how do I navigate through the pages from the browser?

(Sorry lol)

**Comment by JanKrb** [comment]
You will have to enter the register page (most likely http://localhost:8080/register) and create yourself an account. From there you will have to give yourself a role with needed permissions for the frontend you will need:

- roles_get_all
- announcements_get_all
- posts_recent
- posts_store
- categories_store

Which permission, you need for which route, you can check by using postman or checking the permission middleware in the backend's api route.

**Comment by JanKrb** [comment]
From now, you can use the role & permission seeders provided in the alpha branch. See https://github.com/KrbDevelopment/Articly-Backend/commit/1a26fcb976edd206e224d6153250654bbaf36a5b

**Comment by yamanidev** [comment]
I am a bit overwhelmed with the setup, thought all I had to do was debug some CSS/Sass, but hey I like a challenge. I'll try to do my best here.

Thank you for the help!

**Comment by yamanidev** [comment]
I have a problem with passport installation, I'd appreciate it if you could help me with it:
Upon first running the command `php artisan passport:install` I had a similar error that said "could not find driver". I looked it up online and I ran the command `sudo apt-get install php8.0-mysql` which made it go away, though this error appeared, any ideas?

![connection-refused](https://user-images.githubusercontent.com/65856786/131237014-066767ab-874e-48cc-8450-a05d1df486f9.png)

**Comment by JanKrb** [comment]
Please double check, if the mysql server, you provided in the .env is up and running.

**Comment by yamanidev** [comment]
I confirmed that it is running, and the error is gone. Thank you!

What should I provide in the `DB_DATABASE` parameter in the `.env` file though? Because I am getting "Unknown database 'laravel'" error.

**Comment by JanKrb** [comment]
The `DB_DATABASE` parameter specifies the name of the MySQL database. The default value is `laravel`. So you might change that value to a database you wanna use, or just create one called `laravel`.

I guess you will know, but the SQL-statement to create a MySQL database is: `CREATE DATABASE ~database_name~;`
Obviously, you will have to replace `~database_name~` with your database name. When not using the MySQL's root user, you will have to grant privileges to the user for this database.

**Comment by yamanidev** [comment]
I thought I had to use a predefined database by the project? That structures the fields etc.

**Comment by JanKrb** [comment]
No, you will have to create an empty database yourself. By running the migrations `php artisan migrate --seed` you will get all tables created and with the seeders (`--seed` parameter) will get a demo user with roles and permissions created (as soon as you got the version, where it's getting created).

**Comment by yamanidev** [comment]
Okay that makes sense, thank you!

**Comment by yamanidev** [comment]
Okay so I've done the following (verified):
- Migrated the DB successfully.
- Executed all of the commands in README.md in order (both front end and back end).
- Tried entering the credentials that you provided in the seeders, it kept loading.
- Tried creating a new account, same behavior, stuck at loading.

What do you think I am missing here?

P.S: I apologize for taking a while to keep up, it's my first time running a server, using composer, migrating db etc.

**Comment by JanKrb** [comment]
You don't need to apologize, it's fine. :)
Please check out error messages in your dev tools console. Mostlikly on F12 or Crtl + Shift + I.

**Comment by yamanidev** [comment]
Here is the errors log:

`[HMR] Waiting for update signal from WDS... log.js:24
XHRGEThttp://localhost:8080/auth/user
[HTTP/1.1 404 Not Found 4ms]

	
GET
	http://localhost:8080/auth/user
Status404
Not Found
VersionHTTP/1.1
Transferred420 B (148 B size)
Referrer Policystrict-origin-when-cross-origin

    	
    Connection
    	keep-alive
    Content-Length
    	148
    Content-Security-Policy
    	default-src 'none'
    Content-Type
    	text/html; charset=utf-8
    Date
    	Sun, 29 Aug 2021 15:13:05 GMT
    Keep-Alive
    	timeout=5
    X-Content-Type-Options
    	nosniff
    X-Powered-By
    	Express
    	
    Accept
    	application/json, text/plain, */*
    Accept-Encoding
    	gzip, deflate
    Accept-Language
    	en,en-US;q=0.5
    Authorization
    	Bearer null
    Connection
    	keep-alive
    Host
    	localhost:8080
    Referer
    	http://localhost:8080/login
    Sec-Fetch-Dest
    	empty
    Sec-Fetch-Mode
    	cors
    Sec-Fetch-Site
    	same-origin
    User-Agent
    	Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:91.0) Gecko/20100101 Firefox/91.0

XHRPOSThttp://localhost:8080/auth/login
[HTTP/1.1 404 Not Found 2ms]

Uncaught (in promise) TypeError: error.response.data.data is undefined
    handleSubmit Main.vue:180
    promise callback*handleSubmit Main.vue:178
    render/_cache[3]< Main.vue:43
    withModifiers runtime-dom.esm-bundler.js:1205
    callWithErrorHandling runtime-core.esm-bundler.js:154
    callWithAsyncErrorHandling runtime-core.esm-bundler.js:163
    invoker runtime-dom.esm-bundler.js:367
    addEventListener runtime-dom.esm-bundler.js:317
    patchEvent runtime-dom.esm-bundler.js:335
    patchProp runtime-dom.esm-bundler.js:403
    mountElement runtime-core.esm-bundler.js:4791
    processElement runtime-core.esm-bundler.js:4754
    patch runtime-core.esm-bundler.js:4670
    mountChildren runtime-core.esm-bundler.js:4861
    mountElement runtime-core.esm-bundler.js:4782
    processElement runtime-core.esm-bundler.js:4754
    patch runtime-core.esm-bundler.js:4670
    mountChildren runtime-core.esm-bundler.js:4861
    mountElement runtime-core.esm-bundler.js:4782
    processElement runtime-core.esm-bundler.js:4754
    patch runtime-core.esm-bundler.js:4670
    mountChildren runtime-core.esm-bundler.js:4861
    mountElement runtime-core.esm-bundler.js:4782
    processElement runtime-core.esm-bundler.js:4754
    patch runtime-core.esm-bundler.js:4670
    mountChildren runtime-core.esm-bundler.js:4861
    mountElement runtime-core.esm-bundler.js:4782
    processElement runtime-core.esm-bundler.js:4754
    patch runtime-core.esm-bundler.js:4670
    mountChildren runtime-core.esm-bundler.js:4861
    mountElement runtime-core.esm-bundler.js:4782
    processElement runtime-core.esm-bundler.js:4754
    patch runtime-core.esm-bundler.js:4670
    componentEffect runtime-core.esm-bundler.js:5208
    reactiveEffect reactivity.esm-bundler.js:42
    effect reactivity.esm-bundler.js:17
    setupRenderEffect runtime-core.esm-bundler.js:5154
Main.vue:180
    handleSubmit Main.vue:180
    (Async: promise callback)
    handleSubmit Main.vue:178
    handleSubmit self-hosted:1175
    render/_cache[3]< Main.vue:43
    withModifiers runtime-dom.esm-bundler.js:1205
    callWithErrorHandling runtime-core.esm-bundler.js:154
    callWithAsyncErrorHandling runtime-core.esm-bundler.js:163
    invoker runtime-dom.esm-bundler.js:367
    (Async: EventListener.handleEvent)
    addEventListener runtime-dom.esm-bundler.js:317
    patchEvent runtime-dom.esm-bundler.js:335
    patchProp runtime-dom.esm-bundler.js:403
    mountElement runtime-core.esm-bundler.js:4791
    processElement runtime-core.esm-bundler.js:4754
    patch runtime-core.esm-bundler.js:4670
    mountChildren runtime-core.esm-bundler.js:4861
    mountElement runtime-core.esm-bundler.js:4782
    processElement runtime-core.esm-bundler.js:4754
    patch runtime-core.esm-bundler.js:4670
    mountChildren runtime-core.esm-bundler.js:4861
    mountElement runtime-core.esm-bundler.js:4782
    processElement runtime-core.esm-bundler.js:4754
    patch runtime-core.esm-bundler.js:4670
    mountChildren runtime-core.esm-bundler.js:4861
    mountElement runtime-core.esm-bundler.js:4782
    processElement runtime-core.esm-bundler.js:4754
    patch runtime-core.esm-bundler.js:4670
    mountChildren runtime-core.esm-bundler.js:4861
    mountElement runtime-core.esm-bundler.js:4782
    processElement runtime-core.esm-bundler.js:4754
    patch runtime-core.esm-bundler.js:4670
    mountChildren runtime-core.esm-bundler.js:4861
    mountElement runtime-core.esm-bundler.js:4782
    processElement runtime-core.esm-bundler.js:4754
    patch runtime-core.esm-bundler.js:4670
    componentEffect runtime-core.esm-bundler.js:5208
    reactiveEffect reactivity.esm-bundler.js:42
    effect reactivity.esm-bundler.js:17
    setupRenderEffect runtime-core.esm-bundler.js:5154`

**Comment by JanKrb** [comment]
On the first look, you seems to be accessing `/auth/login` instead of `/api/auth/login`. Please add `/api` in your backend url of your frontend .env

**Comment by yamanidev** [comment]
Oh so there is an .env file in the front end too? You should add that to your README.md. 

I will get on with it!

**Comment by JanKrb** [comment]
It's included over there since yesterday (https://github.com/KrbDevelopment/Articly-Frontend/commit/a5dde89e1ee92ab88c3256ae99566a4a7808df1e), but I'll add the `/api` part over there. :)

**Comment by yamanidev** [comment]
So I created the .env file and rebuilt the project, it seems that I am accessing `/api/auth/login` as I am supposed to, but some errors happened:

[HMR] Waiting for update signal from WDS... log.js:24
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at http://localhost:8000/api/auth/user. (Reason: CORS request did not succeed). 2
[Vue Router warn]: The "next" callback was called more than once in one navigation guard when going from "/" to "/". It should be called exactly one time in each navigation guard. This will fail in production. vue-router.esm-bundler.js:72
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at http://localhost:8000/api/auth/user. (Reason: CORS request did not succeed). 2
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at http://localhost:8000/api/auth/login. (Reason: CORS request did not succeed). 2
Uncaught (in promise) TypeError: error.response is undefined
    handleSubmit Main.vue:180
Main.vue:180

**Comment by JanKrb** [comment]
The default backend cors config is optimized for nginx, that's why you have to change it, as you need it. For local development server, you will need to use a config similar to this:

Backend/config/cors.php
`
<?php

return [

    'paths' => ['api/*', 'sanctum/csrf-cookie'],

    'allowed_methods' => ['*'],

    'allowed_origins' => ['*'],

    'allowed_origins_patterns' => [],

    'allowed_headers' => ['*'],

    'exposed_headers' => [],

    'max_age' => 0,

    'supports_credentials' => false,

];
`

In short, `allowed_methods`, `allowed_origins` & `allowed_headers` each needs an item called `'*'`

**Comment by yamanidev** [comment]
In this scenario, do I have to rerun all of the commands in the README.md for the change to take place?

(it's rebuilding as I write this comment, maybe not necessary but just to be sure)

Also, do I need to run the commands for the front end first THEN the back end's second?

**Comment by JanKrb** [comment]
The order of installing front- & backend doesn't matter. The only thing is, the frontend needs the backend to run, to work properly, that's why you might start installing the backend first.

The backend don't need any kind of recompiling. Depending on your machine, you might need to clear the cache to reload configs and routes using `php artisan optimize`. So you don't need to rerun the process of the README. 

The only compiling is `npm run build` or `npm run serve` in the frontend.

Please set the CORS config to the one in this comment https://github.com/KrbDevelopment/Articly-Frontend/issues/8#issuecomment-907861158 and clear your cache. This should fix your error.

**Comment by yamanidev** [comment]
No luck so far...

cors.php file contents:

![cors-file](https://user-images.githubusercontent.com/65856786/131265327-5891ceb0-93d5-4879-a9e9-fc8c103fa912.png)

Error log on console:

[HMR] Waiting for update signal from WDS... log.js:24
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at http://localhost:8000/api/auth/user. (Reason: CORS request did not succeed). 2
[Vue Router warn]: The "next" callback was called more than once in one navigation guard when going from "/" to "/". It should be called exactly one time in each navigation guard. This will fail in production. vue-router.esm-bundler.js:72
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at http://localhost:8000/api/auth/user. (Reason: CORS request did not succeed). 2
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at http://localhost:8000/api/auth/login. (Reason: CORS request did not succeed). 2
Uncaught (in promise) TypeError: error.response is undefined
    handleSubmit Main.vue:180
    promise callback*handleSubmit Main.vue:178
    render/_cache[3]< Main.vue:43
    withModifiers runtime-dom.esm-bundler.js:1205
    callWithErrorHandling runtime-core.esm-bundler.js:154
    callWithAsyncErrorHandling runtime-core.esm-bundler.js:163
    invoker runtime-dom.esm-bundler.js:367
Main.vue:180

**Comment by NinoKrb** [comment]
Have you restarted your backend after your changes? That was sometimes the case with me

> No luck so far...
> 
> cors.php file contents:
> 
> ![cors-file](https://user-images.githubusercontent.com/65856786/131265327-5891ceb0-93d5-4879-a9e9-fc8c103fa912.png)
> 
> Error log on console:
> 
> [HMR] Waiting for update signal from WDS... log.js:24
> Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at http://localhost:8000/api/auth/user. (Reason: CORS request did not succeed). 2
> [Vue Router warn]: The "next" callback was called more than once in one navigation guard when going from "/" to "/". It should be called exactly one time in each navigation guard. This will fail in production. vue-router.esm-bundler.js:72
> Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at http://localhost:8000/api/auth/user. (Reason: CORS request did not succeed). 2
> Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at http://localhost:8000/api/auth/login. (Reason: CORS request did not succeed). 2
> Uncaught (in promise) TypeError: error.response is undefined
> handleSubmit Main.vue:180
> promise callback*handleSubmit Main.vue:178
> render/_cache[3]< Main.vue:43
> withModifiers runtime-dom.esm-bundler.js:1205
> callWithErrorHandling runtime-core.esm-bundler.js:154
> callWithAsyncErrorHandling runtime-core.esm-bundler.js:163
> invoker runtime-dom.esm-bundler.js:367
> Main.vue:180

Have you restarted your backend after your changes? That was sometimes the case with me. If that doesn't help you could try to empty the cache.

**Comment by yamanidev** [comment]
I restarted both the back end (first) and the front end many times as well as empty the cache with `php artisan optimize`, what else am I missing here?

**Comment by NinoKrb** [comment]
Have you tried it with another browser, it is not likely but the Internet says that it can be due to a certain Firefox version.

**Comment by yamanidev** [comment]
I tried Chromium and this is what I've got back:

![chromium](https://user-images.githubusercontent.com/65856786/131265884-33f88f26-79b7-43a5-8dcb-0c5c6164affd.png)

Upon looking up what `net::ERR_CONNETION_REFUSED` means I found that it happens "when the cross origin communication between the backend and frontend is not connected properly"

**Comment by NinoKrb** [comment]
Yes that's right, is it just the error coming or another one?

**Comment by yamanidev** [comment]
I believe it is the same error expressed differently in different browsers, what do you think?

**Comment by NinoKrb** [comment]
No, the "error" is currently still there because you are not logged in and trying to access the system. Please try to log in or register.

**Comment by yamanidev** [comment]
I did try to log in. Here are the logs for both logging in via predefined credentials and when registering an account:
[logging-in-using-predefined-credentials.log](https://github.com/KrbDevelopment/Articly-Frontend/files/7073255/logging-in-using-predefined-credentials.log)
[registering.log](https://github.com/KrbDevelopment/Articly-Frontend/files/7073256/registering.log)

**Comment by NinoKrb** [comment]
Just to understand, your backend is currently started and running on the correct port?

**Comment by yamanidev** [comment]
I followed the commands in README.md, that's all I did.
Or is there other ones I am not aware of?

**Comment by NinoKrb** [comment]
Did you start the backend with Did you start the backend with `php artisan serve`?

**Comment by NinoKrb** [comment]
Starting Laravel development server: http://127.0.0.1:8000
[Sun Aug 29 23:41:55 2021] PHP 8.0.9 Development Server (http://127.0.0.1:8000) started

These messages should appear when your backend starts without problems

**Comment by yamanidev** [comment]
I am sorry no I did not know that I should've done that.

I discovered three things:
1. I am stupid for not knowing that obviously the backend must be also RUNNING!
2. The password I was providing for the preexisting user was wrong. I was copying a long string with weird symbols (which I found on the DB), maybe that's the hashed version of it for security purposes and whatnot? I don't know.
3. I created an account, and I was able to log in!!!!!!!!!!!!!!!

Can you please tell me where I can find the preexisting user's password? Since it already has all the roles/permissions necessary? Thank you very much for your patience with me so far :)!

**Comment by NinoKrb** [comment]
It happens to everyone once ^^. Of course, the passwords of all users are hashed. They can't read the passwords, because they are hashed. You could unhash it with a PHP command or just remove the role from the user.

**Comment by yamanidev** [comment]
I tried providing the value of `'password'` in `Articly-Backend/database/factories/UserFactory.php` to the demo user, didn't work as well.

**Comment by NinoKrb** [comment]
This password is only written there for security, if no password has been given by the user.

**Comment by NinoKrb** [comment]
I just checked, as far as I can see you can't even hide the password, you can only replace it with an artificially created and hashed password.

**Comment by NinoKrb** [comment]
Yes exactly, had not seen your edit sorry

**Comment by yamanidev** [comment]
I am so intrigued about all of this man wow.
I just had an idea: make the user I created "resemble" the preexisting one, I suppose the only column I should be interested in is the `role_id`? 
Do you think I'd get the necessary roles and permissions if I changed my user's column to the same `role_id` as the preexisting one?

**Comment by NinoKrb** [comment]
Exactly, the role ID reflects the stored role with the respective ID. Simply change the number to another available role, then the user should directly belong to another role.

**Comment by NinoKrb** [comment]
So is everything working now, as it should?

**Comment by yamanidev** [comment]
Yes indeed it is I suppose, thank you very much for the help!
I take it that the responsiveness problem is with the Author Tools?

**Comment by NinoKrb** [comment]
Yes, that is also the reason. The problem is currently that the sidebar content is hidden when there is none. So it's never quite sure what should be displayed. For example, the authoring tools are only displayed when the right to them is active. The other things also only if content is available.

**Comment by NinoKrb** [comment]
I would have thought of a kind of banner or footer which is built in the block design. The mobile view for the cell phone works so far.

I have made you a small sketch of what I could imagine
![Sketch](https://user-images.githubusercontent.com/66381853/131267357-07227d8b-e1fd-4adb-8c42-6a0de4558cff.png)

**Comment by yamanidev** [comment]
I am sorry I don't quite understand.
Your problem is that space is acquired to the left of the Authoring Tools even though there's nothing to show?

**Comment by NinoKrb** [comment]
Of course it should be so that if a user has no rights to create things or no content is available, that then the design automatically adapts to it.

**Comment by NinoKrb** [comment]
The problem is that if the page is pulled smaller, then the Responsive design no longer looks good. It should of course look good for the user

**Comment by NinoKrb** [comment]
How you want to implement it is up to you, it should just look good on all devices

**Comment by yamanidev** [comment]
Okay got it, I will get on with it. 
Again, thanks a lot!

**Comment by NinoKrb** [comment]
Have fun, please get in touch if you have any problems. For info, since our system is not 100% finished and you do not have the latest version, it may be that some functions do not work.

**Comment by yamanidev** [comment]
Before I can do anything, the page's content must all be present and in place.
I want to have first the page filled 100% (full permissions/roles) with all the content present. When we work on the responsiveness of that page, having fewer roles/permissions would be easier to manage. Do you see what I mean?

Also, I think communicating via DMs would be much suitable for exchanges of the sort.

**Comment by NinoKrb** [comment]
Yes, you can fill the page by creating and authorizing the necessary posts. Then you only need at least one announcement to display the slider. We can also write via DM, what would you prefer as a platform?

**Comment by yamanidev** [comment]
I understand, I just want to know all of the page's content.
What I know now there is: 
- Categories section
- Announcement section
- Author Tools
- Recent posts
If there is any other section, please let me know.

I don't have a preference for platforms. I noticed you have a Twitter account, is it okay by you to use that?

**Comment by NinoKrb** [comment]
I don't use Twitter much, but you are welcome to write to me there.

Exactly, so the entire content is as you described it, I have here another screenshot of how it would look complete.


![Screenshot](https://user-images.githubusercontent.com/66381853/131267949-723a1a66-26db-460f-a07a-a8ac13ecbff4.png)

**Comment by yamanidev** [comment]
Okay then great. I will contact you for any further information.
Thanks a lot, have a great day/night!

**Comment by JanKrb** [comment]
Is there any update about this?

**Comment by yamanidev** [comment]
I am still working on it.
I will send you some remarks on Twitter very soon though.

**Comment by NinoKrb** [comment]
That sounds great, have fun!

**Comment by yamanidev** [comment]
Hello,
Unfortunately I won't be able to fiddle in the project as I previously did due to an upcoming exam of mine. If your project does not have a deadline for completion or anything of the sort, I'd still be happy to resume on this issue (after around 20-ish days).
Thank you for your time, I appreciate it!

**Comment by NinoKrb** [comment]
Hey,
we would just take it from there since we would need it by next week Tuesday. But you are welcome to help for next tasks with pleasure. The project you have set up so far then the next time will certainly work better.

Best regards

**Comment by NinoKrb** [comment]
Solved in https://github.com/KrbDevelopment/Articly-Frontend/commit/d0eaaf6d4483e46813da1ca5e0150a672bac448d & https://github.com/KrbDevelopment/Articly-Frontend/commit/c382325e6cbe632f3ff488cf81afd52d79bdda43
